app/services/media_storage/azure_adapter.rb

```diff
@@ -60,6 +63,11 @@ def put_path(path, opts={})
       ).to_s
     end
 
+    # Returns required headers for making a put request to the Blob Service
+    def headers_for_direct_upload
```

Comment:
where will this be called as a public method? 

I assume these are required for by a client when using a signed PUT URL. If so perhaps they should be returned as headers on the put URL response? 

Alternatively if these aren't dynamic we can add these in as code as well on the uploading clients themselves (PFE, python client etc). That way we provide the client with the info it needs via response headers but also ensure we have these in use on our trusted clients.

Reply:
> I assume these are required for by a client when using a signed PUT URL. If so perhaps they should be returned as headers on the put URL response?

@camallen yep the header is required when using a PUT URL whether it is signed or not (i.e. required for both public and private uploads). What do you mean when you say 'they should be returned as headers on the PUT URL response'? Currently, we send over the PUT URL in the `src` attribute of the serialized medium object- my thought was that we could add a `headers` attribute on the medium serializer that the client could access. Is there a different way to do it than that?
https://github.com/zooniverse/panoptes/blob/85f26dcaaf4b8788a802874d28521c9eed0b162f/app/serializers/medium_serializer.rb#L19

>Alternatively if these aren't dynamic we can add these in as code as well on the uploading clients themselves (PFE, python client etc).

Indeed, the headers are not dynamic. So you're saying we could just add code for the header _just_ in our uploading clients (PFE, python client etc) and not make any changes to the panoptes backend? My only concern with this is that we only want to be adding the header if we're talking to azure storage, and technically we dont want the client to need to worry about whether it is uploading to azure or to aws.

Reply:
> What do you mean when you say 'they should be returned as headers on the PUT URL response'?
As literal response headers, not in the response object. https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers

I like the fact you've copied the ActiveStorage code here, checkout how it's used in the HTTP response object as headers, https://github.com/rails/rails/blob/855de1a7004a6cb87b7f637b2e94edda5f3e4e56/activestorage/app/controllers/active_storage/direct_uploads_controller.rb#L20

I believe our clients do most of this work themselves, e.g. https://github.com/zooniverse/panoptes-python-client/blob/7f3f4833adbd89bb60e9965b4a747ef7d552283a/panoptes_client/subject.py#L185

Providing support for suggested client upload headers from the API repsonse is a good idea 👍